tests/unit/s2n_ktls_io_test.c

```diff
@@ -27,18 +27,30 @@ S2N_RESULT s2n_ktls_set_control_data(struct msghdr *msg, char *buf, size_t buf_s
 S2N_RESULT s2n_ktls_get_control_data(struct msghdr *msg, int cmsg_type, uint8_t *record_type);
 
 /* Mock implementation used for validating failure behavior */
-struct s2n_test_ktls_io_fail {
-    size_t invoked_count;
+struct s2n_test_ktls_io_fail_ctx {
     size_t errno_code;
+    ssize_t return_value;
```

Comment:
I don't think return_value belongs here at the moment. The callback is called "..._fail", so the return value should always be 0. You'll either want a separate callback/context, or to do some renames. Separate might be better-- you'd avoid having to manually set "-1" everywhere you expect an error, which is a much more common case in your tests than EOF.